### PR TITLE
Disputes: Add amounts to reward event

### PR DESCRIPTION
**Pull request by bingen**

### CODE REVIEW

File: contracts/disputes/DisputeManager.sol [hunk 1]
```diff
@@ -1224,6 +1229,24 @@ contract DisputeManager is ControlledRecoverable, ICRVotingOwner, IDisputeManage
...
+    * @param _jurorState Juror with state which the weighted amount is computed for
+    * @param _amount Amount to be weighted
+    * @return Weighted amount for a juror in a round in relation to total amount of coherent jurors
+    */
+    function _getRoundWeightedAmount(
+        AdjudicationRound storage _round,
+        JurorState storage _jurorState,
+        uint256 _amount
+    )
+        internal
```
Comment (facuspagnuolo) [inline contracts/disputes/DisputeManager.sol]: let's mark this one as `view` and place it along with the `internal view` functions

File: contracts/disputes/DisputeManager.sol [hunk 2]
```diff
@@ -1224,6 +1229,24 @@ contract DisputeManager is ControlledRecoverable, ICRVotingOwner, IDisputeManage
...
+    * @param _round Round which the weighted amount is computed for
+    * @param _jurorState Juror with state which the weighted amount is computed for
+    * @param _amount Amount to be weighted
+    * @return Weighted amount for a juror in a round in relation to total amount of coherent jurors
+    */
+    function _getRoundWeightedAmount(
+        AdjudicationRound storage _round,
+        JurorState storage _jurorState,
+        uint256 _amount
+    )
```
Comment (facuspagnuolo) [inline contracts/disputes/DisputeManager.sol]: Nit: I think we can avoid breaking these into multiple lines

File: contracts/disputes/DisputeManager.sol [hunk 3]
```diff
@@ -432,32 +432,37 @@ contract DisputeManager is ControlledRecoverable, ICRVotingOwner, IDisputeManage
...
         // Set the lock for final round
         if (!_isRegularRound(_roundId, config)) {
             // Round end term ID (as it's final there's no draft delay nor appeal) plus the lock period
             DisputesConfig memory disputesConfig = config.disputes;
-            uint64 finalRoundLockTermId = round.draftTermId +
-                disputesConfig.commitTerms + disputesConfig.revealTerms + disputesConfig.finalRoundLockTerms;
-            jurorsRegistry.lockWithdrawals(_juror, finalRoundLockTermId);
+            jurorsRegistry.lockWithdrawals(
+                _juror,
+                round.draftTermId + disputesConfig.commitTerms + disputesConfig.revealTerms + disputesConfig.finalRoundLockTerms
```
Comment (facuspagnuolo) [inline contracts/disputes/DisputeManager.sol]: this inline was to avoid a stack-too-deep issue?

File: contracts/disputes/DisputeManager.sol [hunk 4]
```diff
@@ -432,32 +432,37 @@ contract DisputeManager is ControlledRecoverable, ICRVotingOwner, IDisputeManage
...
 
-        // Note that the number of coherent jurors has to be greater than zero since we already ensured the juror has voted in favor of the
-        // final ruling, therefore there will be at least one coherent juror and divisions below are safe.
-        uint256 coherentJurors = round.coherentJurors;
         uint256 collectedTokens = round.collectedTokens;
         IJurorsRegistry jurorsRegistry = _jurorsRegistry();
 
         // Distribute the collected tokens of the jurors that were slashed weighted by the winning jurors. Note that we are penalizing jurors
         // that refused intentionally their vote for the final round.
+        uint256 collectedTokensReward;
```
Comment (facuspagnuolo) [inline contracts/disputes/DisputeManager.sol]: what do you think about `rewardTokens` simply?

File: contracts/disputes/DisputeManager.sol [hunk 5]
```diff
@@ -140,7 +140,7 @@ contract DisputeManager is ControlledRecoverable, ICRVotingOwner, IDisputeManage
     event RulingAppealConfirmed(uint256 indexed disputeId, uint256 indexed roundId, uint64 indexed draftTermId, uint256 jurorsNumber);
     event RulingComputed(uint256 indexed disputeId, uint8 indexed ruling);
     event PenaltiesSettled(uint256 indexed disputeId, uint256 indexed roundId, uint256 collectedTokens);
-    event RewardSettled(uint256 indexed disputeId, uint256 indexed roundId, address juror);
+    event RewardSettled(uint256 indexed disputeId, uint256 indexed roundId, address juror, uint256 collectedTokens, uint256 jurorFees);
```
Comment (facuspagnuolo) [inline contracts/disputes/DisputeManager.sol]: I would name these simply `tokens` and `fees`, WDYT?

File: contracts/disputes/DisputeManager.sol (hunk 3 above)
Comment (bingen) [reply contracts/disputes/DisputeManager.sol]: Yes

File: contracts/disputes/DisputeManager.sol (hunk 2 above)
Comment (bingen) [reply contracts/disputes/DisputeManager.sol]: No, we can't, it's 153 chars long.

**Pull request merged**